### PR TITLE
[google-nik-collection] Is free now

### PR DESCRIPTION
**Pull request by renard**
See https://plus.google.com/+NikCollection/posts/AFGsG2Di7EK

Signed-off-by: Sébastien Gross <seb•ɑƬ•chezwam•ɖɵʈ•org>

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Actually, this needs is missing `uninstall`. There are [scripts](https://github.com/caskroom/homebrew-cask/tree/master/developer/bin) and [documentation](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/uninstall.md) geared for that.

Could you try making a stanza for it? If not, we should likely remove this.

**Comment by renard** [comment]
@vitorgalvao they provide an uninstall app (`/Applications/Nik Collection/Uninstaller.app`) is that a good way to do?

File: Casks/google-nik-collection.rb [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+             }
+
+  zap delete: [
+        '~/Library/Google/GoogleSoftwareUpdate/Actives/com.google.nikcollection',
+        '~/Library/Preferences/Google/Nik Collection',
+        '~/Library/Preferences/Google/Nik Collection Installer Logs',
+        '/Library/Preferences/Google/',
+        '/var/root/Library/Preferences/Google/Nik Collection Installer Logs',
+      ]
+
```
Comment (vitorgalvao) [inline Casks/google-nik-collection.rb]: Extraneous line.

**Comment by vitorgalvao** [comment]
> is that a good way to do?

No, because it calls a GUI (I’m guessing).

**Comment by renard** [comment]
@vitorgalvao I guess there is no other way except removing files manually.

OTH the install procedure is manual: you have to explicitly call the `Nik Collection.app`

**Comment by vitorgalvao** [comment]
> I guess there is no other way except removing files manually.

Yes, we need to basically regress the changes made by the installer.

> the install procedure is manual

It is. But we don’t like uninstalls to be. Those are easy to miss.

**Comment by renard** [comment]
@vitorgalvao Travis and rubocop are really pissing me off. I don't understand the meaning of that error:

```
C: 14: 13: Align the elements of a hash literal if they span more than one line.
C: 19: 13: Align the elements of a hash literal if they span more than one line.
```

I cannot run rubocop on my own machine got a lot of errors:

```
(ruby) renard@scrat:~/Src/homebrew-cask $ rubocop Casks/google-nik-collection.rb
Inspecting 1 file


0 files inspected, no offenses detected
/Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- /Users/renard/Src/homebrew-cask/Casks/rubocop-cask (LoadError)
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/config_loader.rb:40:in `block in load_file'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/config_loader.rb:39:in `each'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/config_loader.rb:39:in `load_file'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/config_loader.rb:99:in `configuration_from_file'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/config_store.rb:41:in `for'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/result_cache.rb:114:in `file_checksum'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/result_cache.rb:67:in `initialize'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/runner.rb:82:in `new'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/runner.rb:82:in `process_file'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/runner.rb:59:in `block in inspect_files'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/runner.rb:57:in `each'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/runner.rb:57:in `inspect_files'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/runner.rb:35:in `run'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/lib/rubocop/cli.rb:30:in `run'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/bin/rubocop:14:in `block in <top (required)>'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
	from /Users/renard/.ry/lib/ry/rubies/2.2.4/lib/ruby/gems/2.2.0/gems/rubocop-0.38.0/bin/rubocop:13:in `<top (required)>'
	from /Users/renard/.ry/lib/ry/current/bin/rubocop:23:in `load'
	from /Users/renard/.ry/lib/ry/current/bin/rubocop:23:in `<main>'
(ruby) renard@scrat:~/Src/homebrew-cask $
```

Just hope the Cask is ok with you

File: Casks/google-nik-collection.rb [hunk 2]
```diff
@@ -1,11 +1,33 @@
...
+                # Remove installed support in Lightroom
+                '~/Library/Application Support/Adobe/Lightroom/External Editor Presets/AnalogEfexPro2.lrtemplate',
+                '~/Library/Application Support/Adobe/Lightroom/External Editor Presets/ColorEfexPro4.lrtemplate',
+                '~/Library/Application Support/Adobe/Lightroom/External Editor Presets/Dfine2.lrtemplate',
+                '~/Library/Application Support/Adobe/Lightroom/External Editor Presets/SharpenerPro3_1_RAWPresharpener.lrtemplate',
+                '~/Library/Application Support/Adobe/Lightroom/External Editor Presets/SharpenerPro3_2_OutputSharpener.lrtemplate',
+                '~/Library/Application Support/Adobe/Lightroom/External Editor Presets/SilverEfexPro2.lrtemplate',
+                '~/Library/Application Support/Adobe/Lightroom/External Editor Presets/Viveza2.lrtemplate',
+                # End of lightroom plugins
+                '/Library/Preferences/Google/',
```
Comment (vitorgalvao) [inline Casks/google-nik-collection.rb]: Won’t this possibly contain other Google-related stuff?

File: Casks/google-nik-collection.rb [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
-  url 'https://dl.google.com/dl/edgedl/photos/nikcollection-demo.dmg'
+  url "https://dl.google.com/edgedl/photos/nikcollection-full-#{version}.dmg"
   name 'Google Nik Collection'
   homepage 'https://www.google.com/nikcollection/'
-  license :commercial
+  license :closed
 
   installer manual: 'Nik Collection.app'
+
+  # Nik Collection only provide an Uninstaller.app which runs a GUI.
```
Comment (vitorgalvao) [inline Casks/google-nik-collection.rb]: These comments should go.

File: Casks/google-nik-collection.rb [hunk 4]
```diff
@@ -1,11 +1,33 @@
...
+  uninstall delete: [
+                      '/Applications/Nik Collection',
+                    ]
+
+  zap delete: [
+                '~/Library/Google/GoogleSoftwareUpdate/Actives/com.google.nikcollection',
+                '~/Library/Preferences/Google/Nik Collection',
+                '~/Library/Preferences/Google/Nik Collection Installer Logs',
+                # Remove installed support in Lightroom
+                '~/Library/Application Support/Adobe/Lightroom/External Editor Presets/AnalogEfexPro2.lrtemplate',
```
Comment (vitorgalvao) [inline Casks/google-nik-collection.rb]: These Lightroom things should be part of `uninstall`.

**Comment by renard** [comment]
@vitorgalvao Do you want me to collapse all these commits?

File: Casks/google-nik-collection.rb (hunk 2 above)
Comment (renard) [inline Casks/google-nik-collection.rb]: As far as I know, no. But I change it to be more selective, which can result to an empty directory.

File: Casks/google-nik-collection.rb (hunk 3 above)
Comment (renard) [inline Casks/google-nik-collection.rb]: What about comments. Should we leave some to help remember why some things are done?

**Comment by vitorgalvao** [comment]
Yes, please.

File: Casks/google-nik-collection.rb (hunk 3 above)
Comment (vitorgalvao) [inline Casks/google-nik-collection.rb]: No, because this is standard.

File: Casks/google-nik-collection.rb (hunk 2 above)
Comment (vitorgalvao) [inline Casks/google-nik-collection.rb]: Use `:rmdir` on it, then, as it only acts on empty directories.

**Comment by renard** [comment]
@vitorgalvao done.
Thanks for your patience and your helpful support.

**Comment by vitorgalvao** [comment]
Thank you for all the availability and changes (and contribution, naturally).

**Pull request merged**